elections/models.py

```diff
@@ -194,7 +194,10 @@ def group_and_order_elections(cls, include_posts=False,
         last_current = None
         for election in qs:
             current_index = 1 - int(election.current)
-            roles = result[current_index]['roles']
+            date = election.election_date
+            if date not in result[current_index]['dates']:
+                result[current_index]['dates'][date] = []
+            roles = result[current_index]['dates'][date]
```

Comment:
These 4 lines could be replaced by 1, if you do:

```python
roles = result[current_index]['dates'].setdefault(election.election_date, [])
```

Or actually, it'd be probably be clearer (+ keep under the line length limit) to still have `date = election.election_date` as a separate line.